<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Document</title>
  </head>
  <body>
    <script>
              // let p1 = Promise.resolve('1')
              // let p2 = Promise.resolve('2')
              // let p3 = Promise.resolve('3')


              // Promise.all([p1,p2,p3])
              // .then((value) => console.log(value))
              // .catch((err)=> console.log(err)
              // )

          // let myPromise =  Promise.all([
          //         new Promise( resolve => setTimeout( resolve(1),3000)),
          //         new Promise( reject => setTimeout( reject('Error'),2000)),
          //         new Promise( resolve => setTimeout( resolve(2),1000)),
          //     ])

          //     myPromise.then(alert)
      //     let myPromise =  Promise.race([
      //             new Promise( resolve => setTimeout( resolve(1),3000)),

      //               new Promise( reject => setTimeout( reject('Error'),2000)),

      //             new Promise( resolve => setTimeout( resolve(2),1000)),
      //         ])

      //      myPromise.then(result => console.log(result)
      //      )

      //  let p1 = new Promise((resolve)=> setTimeout(resolve(1),2000))
      //  let p2 = new Promise((resolve)=> setTimeout(resolve(2),1000))
      //     Promise.race([p1,p2]).then(value  =>  console.log(value)
      //     )



    //   let promise = Promise.any([
        
    //   new Promise((resolve,reject)=> setTimeout(reject(new
    //   Error("Its an Error!")),1000)),

    //           new Promise((resolve,reject)=>
    //   setTimeout(resolve(2),4000)),

    //      new Promise((resolve,reject)=>
    //   setTimeout(resolve(1),3000)),

     

    //   ])
    //   promise.then( value => console.log(value)
    //   )


   let p1 = Promise.reject('1')
   let p2 = Promise.reject('2')
   let p3 = Promise.reject('3')


              Promise.any([p1,p2,p3])
              .then((value) => console.log(value))
              .catch((err)=> console.log(err)
              )

    </script>
  </body>
</html>
